diagrams/unifiedPushMobile00.puml

```diff
@@ -0,0 +1,41 @@
+@startuml unifiedPushMobile00
+actor User
+participant MobileApp
+participant API
+database Postgres
+participant NtfyMobileApp
+participant UPServer
+
+User -> MobileApp : opens the app
+MobileApp -> User : asks the user to accept notifications
+User -> MobileApp : accepts notifications
+MobileApp -> User : enables notifications
+User -> MobileApp : tries to subscribe to notifications for a podcast
+MobileApp -> User : prompts the user that notifications are a \npremium-only feature and to please login
+User -> MobileApp : user logs in, and tries to subscribe \nto notifications for a podcast again
+MobileApp -> User : shows the UnifiedPush setup modal
+User -> MobileApp : inputs their UP endpoint hostname \nand presses the Register button
+MobileApp -> API : sends UP hostname in \nan authenticated request
+API -> API : creates UP salted_topic_id
+API -> Postgres : saves the UP hostname+salted_topic_id \nwith ManyToOne to User table \nin the unifiedPushToken table
```

Comment:
And the API doesn't create the topic, the app receives it from the distributor, and it uses that topic. You have to treat the hostname+topic as one single URL, they're not separable.

Reply:
If you need to, you can use "instances" during (MobileApp -> Distributor) registration to have an endpoint per subject. You can also get a single endpoint for the app and push a content that will help to identify the podcast subject

Reply:
> How would this properly scale?
> What are the scaling limits?
> Would/could we be blocked by a commonly used U.P. server?

1. Matrix servers send hundreds of messages per device per day, and we have fixed most rate limiting problems. 
2. Performance-wise, it shouldn't be a big deal because of connection reuse - as long as the requests are batched properly, node (or another runtime) should automatically send all the HTTP requests over one connection.
3. Also, WebPush, the standard for push notifications to web browsers, uses the exact same model (unique endpoint for each device) and all services that push to web browsers use it.